crawl-ref/source/dat/des/variable/large_themed.des

```diff
@@ -2316,3 +2316,85 @@ xxxx-xx..xx....xx-xxx-x
 xxxxxxxxx-x.....xxxxxxx
 xxxxxxxxxxx.@@@..xxxxxx
 ENDMAP
+
+NAME:   dreamdust_cave_troll
+TAGS:   no_monster_gen
+ORIENT: northwest
+DEPTH:  D:10-, Lair
+NSUBST: - = 1 / 2=? / .
+KFEAT:  ^ = web
```

Comment:
web trap doesn't feel quite right here, since that's something we always use with spider-type monsters. I recommend use of net traps, and try to place the net trap such that the troll might step on it as they chase the player. Note that it's also possible to only have a chance to place the trap on each glyph or on certain glyphs, if you're concerned it's too nasty.